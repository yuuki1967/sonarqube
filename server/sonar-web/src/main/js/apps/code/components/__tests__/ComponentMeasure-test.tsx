/*
 * SonarQube
 * Copyright (C) 2009-2019 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import { shallow } from 'enzyme';
import ComponentMeasure from '../ComponentMeasure';

const METRIC = { id: '1', key: 'coverage', type: 'PERCENT', name: 'Coverage' };
const LEAK_METRIC = { id: '2', key: 'new_coverage', type: 'PERCENT', name: 'Coverage on New Code' };
const COMPONENT = { key: 'foo', name: 'Foo', qualifier: 'TRK' };
const COMPONENT_MEASURE = {
  ...COMPONENT,
  measures: [{ value: '3.0', periods: [{ index: 1, value: '10.0' }], metric: METRIC.key }]
};

it('renders correctly', () => {
  expect(
    shallow(<ComponentMeasure component={COMPONENT_MEASURE} metric={METRIC} />)
  ).toMatchSnapshot();
});

it('renders correctly for leak values', () => {
  expect(
    shallow(
      <ComponentMeasure
        component={{
          ...COMPONENT,
          measures: [
            { value: '3.0', periods: [{ index: 1, value: '10.0' }], metric: LEAK_METRIC.key }
          ]
        }}
        metric={LEAK_METRIC}
      />
    )
  ).toMatchSnapshot();
});

it('renders correctly when no measure found', () => {
  expect(shallow(<ComponentMeasure component={COMPONENT} metric={METRIC} />)).toMatchSnapshot();
});
