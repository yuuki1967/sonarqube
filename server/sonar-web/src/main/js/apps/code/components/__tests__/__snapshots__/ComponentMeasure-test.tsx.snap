// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Measure
  metricKey="coverage"
  metricType="PERCENT"
  value="3.0"
/>
`;

exports[`renders correctly for leak values 1`] = `
<Measure
  metricKey="new_coverage"
  metricType="PERCENT"
  value="10.0"
/>
`;

exports[`renders correctly when no measure found 1`] = `<span />`;
